packages/mattermost-redux/src/reducers/entities/threads/counts.ts

```diff
@@ -139,7 +149,26 @@ export function countsIncludingDirectReducer(state: ThreadsState['counts'] = {},
     case ChannelTypes.RECEIVED_CHANNEL_DELETED:
     case ChannelTypes.LEAVE_CHANNEL:
         return handleLeaveChannel(state, action, extra);
+    case ThreadTypes.RECEIVED_UNREAD_THREADS: {
+        if (isEqual(state, action)) {
+            return state;
+        }
+
+        const counts = state[action.data.team_id] ?? {};
+        return {
+            ...state,
+            [action.data.team_id]: {
+                ...counts,
+                total_unread_threads: action.data.total_unread_threads,
```

Comment:
If these are numbers, wouldn't it be more readable to do something like:
```
const team = state[action.data.team_id];
if (action.data.total_unread_threads !== team.total_unread_threads || action.data.total_unread_mentions !== team.total_unread_mentions) {
    ...
}
```

Reply:
@larkox, that's true, and also made me realize that there is another bug in what I wrote.
Pushed a new commit